scripts/publish_npm.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# Initialize variables
+new_version=""
+simulate=false
+build=false
+update=false
+
+# Function to compare version numbers
+version_gt() { test "$(printf '%s\n' "$@" | sort -V | head -n 1)" != "$1"; }
+
+# Parse command line arguments
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        -v|--version) new_version="$2"; shift ;;
+        --simulate) simulate=true ;;
+        --build) build=true ;;
+        --update) update=true ;;  
+        *) echo "Unknown parameter passed: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+# Update and install dependencies if --update is set
+if [ "$update" = true ]; then
+    echo "Updating repository and installing dependencies..."
+    git pull
+    yarn install
+fi
+
+ # Build the project if --build is set
+if [ "$build" = true ]; then
+  echo "Building project..."
+  yarn build
+fi
+
+# Loop through all directories
+find . -type d | while read dir; do
+  # Check if both package.json and .export-npm exist
+  if [[ -f "$dir/package.json" && -f "$dir/.export-npm" ]]; then
+    # Change to the directory
+    cd "$dir"
+
+    # Read the package name from package.json
+    package_name=$(jq -r '.name' package.json)
+
+    # Get the current version from npm
+    current_npm_version=$(npm view "$package_name" version 2>/dev/null)
+
+    # Check if new version is greater than current version
+    if [ -n "$new_version" ] && version_gt "$new_version" "$current_npm_version"; then
```

Comment:
Do we have to always release every package when using this script ?

Reply:
For now, the script is for releasing all. We may extend it late for more detailed features.